STAR-Delegation/CSR-template/template-schema.cddl

```diff
@@ -0,0 +1,79 @@
+csr-template-schema = {
+  ? keyTypes: [ 1* $keyType ]
+  ? subject: distinguishedName
+  ? extensions: extensions
+}
+
+mandatory-wildcard = "**"
+optional-wildcard = "*"
+wildcard = mandatory-wildcard / optional-wildcard
+
+; regtext matches all text strings but "*" and "**"
+regtext = text .regexp "([^\*].*)|([\*][^\*].*)|([\*][\*].+)"
+
+regtext-or-wildcard = regtext / wildcard
+
+distinguishedName = {
+  ? country: regtext-or-wildcard
+  ? stateOrProvince: regtext-or-wildcard
+  ? locality: regtext-or-wildcard
+  ? organization: regtext-or-wildcard
+  ? organizationalUnit: regtext-or-wildcard
+  ? emailAddress: regtext-or-wildcard
+  ? commonName: regtext-or-wildcard
+}
+
+$keyType /= rsaKeyType
+$keyType /= ecKeyType
+
+rsaKeyType = {
+  ? PublicKeyType: "RSA"
```

Comment:
I don't think it makes sense for PublicKeyType to be optional

Reply:
Right. And we should add `required` array liberally to the JSON schema, to make object properties mandatory.